commit a68b2c2c8c428689ec838c228555fea1ed3d07bb
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Thu May 11 16:58:16 2017 +0100

    BCM270X_DT: Tidy up mmc, sdhost, sdio overlays
    
    The mmc, sdhost, sdio and sdio-1bit overlays had a few
    anachronisms and oddities which were overdue for fixing.
    The new versions should be functionally equivalent.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/overlays/mmc-overlay.dts b/arch/arm/boot/dts/overlays/mmc-overlay.dts
index d32b02ca01ce..64f1ff6684b1 100644
--- a/arch/arm/boot/dts/overlays/mmc-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mmc-overlay.dts
@@ -21,6 +21,7 @@
 			mmc_pins: mmc_pins {
 				brcm,pins = <48 49 50 51 52 53>;
 				brcm,function = <7>; /* alt3 */
+				brcm.pull = <0 2 2 2 2 2>;
 			};
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/sdhost-overlay.dts b/arch/arm/boot/dts/overlays/sdhost-overlay.dts
index a431177a1c9d..de3d1b0a5e40 100644
--- a/arch/arm/boot/dts/overlays/sdhost-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdhost-overlay.dts
@@ -11,7 +11,6 @@
 		frag0: __overlay__ {
 			brcm,overclock-50 = <0>;
 			brcm,pio-limit = <1>;
-			brcm,debug-flags = <0>;
 			status = "okay";
 		};
 	};
diff --git a/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts b/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts
index 46d453859b31..0b7f2c6f2603 100644
--- a/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdio-1bit-overlay.dts
@@ -1,11 +1,12 @@
-/* Enable 1-bit SDIO from MMC interface via GPIOs 22-25. Includes sdhost overlay. */
+/dts-v1/;
+/plugin/;
 
-/include/ "sdhost-overlay.dts"
+/* Enable 1-bit SDIO from MMC interface via GPIOs 22-25. Includes sdhost overlay. */
 
 /{
 	compatible = "brcm,bcm2708";
 
-	fragment@3 {
+	fragment@0 {
 		target = <&mmc>;
 		sdio_mmc: __overlay__ {
 			compatible = "brcm,bcm2835-mmc";
@@ -18,12 +19,12 @@
 		};
 	};
 
-	fragment@4 {
+	fragment@1 {
 		target = <&gpio>;
 		__overlay__ {
 			sdio_pins: sdio_pins {
 				brcm,pins = <22 23 24 25>;
-				brcm,function = <7 7 7 7>; /* ALT3 = SD1 */
+				brcm,function = <7>; /* ALT3 = SD1 */
 				brcm,pull = <0 2 2 2>;
 			};
 		};
diff --git a/arch/arm/boot/dts/overlays/sdio-overlay.dts b/arch/arm/boot/dts/overlays/sdio-overlay.dts
index 398bd812c716..215d5e3e8a8c 100644
--- a/arch/arm/boot/dts/overlays/sdio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sdio-overlay.dts
@@ -1,11 +1,12 @@
-/* Enable SDIO from MMC interface via GPIOs 22-27. Includes sdhost overlay. */
+/dts-v1/;
+/plugin/;
 
-/include/ "sdhost-overlay.dts"
+/* Enable SDIO from MMC interface via GPIOs 22-27. Includes sdhost overlay. */
 
 /{
 	compatible = "brcm,bcm2708";
 
-	fragment@3 {
+	fragment@0 {
 		target = <&mmc>;
 		sdio_mmc: __overlay__ {
 			pinctrl-names = "default";
@@ -17,12 +18,12 @@
 		};
 	};
 
-	fragment@4 {
+	fragment@1 {
 		target = <&gpio>;
 		__overlay__ {
 			sdio_pins: sdio_pins {
 				brcm,pins = <22 23 24 25 26 27>;
-				brcm,function = <7 7 7 7 7 7>; /* ALT3 = SD1 */
+				brcm,function = <7>; /* ALT3 = SD1 */
 				brcm,pull = <0 2 2 2 2 2>;
 			};
 		};
